test/views/coronavirus_local_restrictions/results_html_test.rb

```diff
@@ -0,0 +1,169 @@
+      country_name = "Scotland"
+
+      render_devolved_nation_guidance(postcode, country_name)
+
+      assert_includes rendered, I18n.t("coronavirus_local_restrictions.results.devolved_nations.scotland.guidance.label")
+    end
+
+    test "rendering results for a Northern Irish postcode" do
+      postcode = "BT48 7PX"
+      country_name = "Northern Ireland"
+
+      render_devolved_nation_guidance(postcode, country_name)
+
+      assert_includes rendered, I18n.t("coronavirus_local_restrictions.results.devolved_nations.northern_ireland.guidance.label")
+    end
+  end
+
+  describe "future restrictions" do
+    test "rendering restrictions changing from level one to level two" do
+      render_tier_results(current_alert_level: 1, future_alert_level: 2)
+      assert_includes rendered, I18n.t("coronavirus_local_restrictions.results.level_one.changing_alert_level", area: area)
+      assert_includes rendered, I18n.t("coronavirus_local_restrictions.results.future.level_two.alert_level", area: area)
+    end
+
+    test "rendering restrictions changing from level two to level three" do
+      render_tier_results(current_alert_level: 2, future_alert_level: 3)
+      assert_includes rendered, I18n.t("coronavirus_local_restrictions.results.level_two.changing_alert_level", area: area)
+      assert_includes rendered, I18n.t("coronavirus_local_restrictions.results.future.level_three.alert_level", area: area)
+    end
+
+    test "rendering restrictions changing from level three to level four" do
+      render_tier_results(current_alert_level: 3, future_alert_level: 4)
+      assert_includes rendered, I18n.t("coronavirus_local_restrictions.results.level_three.changing_alert_level", area: area)
+      assert_includes rendered, I18n.t("coronavirus_local_restrictions.results.future.level_four.alert_level", area: area)
+    end
+  end
+
+  describe "out of date restrictions" do
+    test "rendering an out of date warning on the tier one page" do
+      render_tier_results_out_of_date_warning(current_alert_level: 1)
+
+      assert_includes rendered, I18n.t("coronavirus_local_restrictions.out_of_date_warning")
+    end
+
+    test "rendering an out of date warning on the tier two page" do
+      render_tier_results_out_of_date_warning(current_alert_level: 2)
+
+      assert_includes rendered, I18n.t("coronavirus_local_restrictions.out_of_date_warning")
+    end
+
+    test "rendering an out of date warning on the tier three page" do
+      render_tier_results_out_of_date_warning(current_alert_level: 3)
+
+      assert_includes rendered, I18n.t("coronavirus_local_restrictions.out_of_date_warning")
+    end
+
+    test "rendering an out of date warning on the tier four page" do
+      render_tier_results_out_of_date_warning(current_alert_level: 4)
+
+      assert_includes rendered, I18n.t("coronavirus_local_restrictions.out_of_date_warning")
+    end
+  end
+
+  def render_tier_results(current_alert_level: nil, future_alert_level: nil)
```

Comment:
As there is a lot of similarity between these methods I wonder if we could have one method that covers them all?

For example:

```
def render_results_view(local_restriction: {}, out_of_date: false)
  local_restriction = { postcode: "E1 8QS" }.merge(local_restriction)
  @search = PostcodeLocalRestrictionSearch.new(local_restriction[:postcode])
  stub_local_restriction(**local_restriction)
  view.stubs(:out_of_date?).returns(out_of_date)

  render template: "coronavirus_local_restrictions/results"
end
```

And then you'd have to adapt the calls to pass a local_restriction argument with a hash:

```
render_results_view(local_restriction: { current_alert_level: 4 })
render_results_view(local_restriction: { country_name: "Northern Ireland" })
```

Reply:
ah that is way better!! thank you! I have refactored to use your suggestion.